[ie/tver] Extract release date or year

fireattack opened the pull request:
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

<!--

Explanation of your *pull request* in arbitrary form goes here. Please **make sure the description explains the purpose and effect** of your *pull request* and is worded well enough to be understood. Provide as much **context and examples** as possible

-->

Extract release date (or year, if not in current year) from onair_label, as it could be drastically different from upload_date givne by underlying brightcove extractor.

<details open><summary>Template</summary> <!-- OPEN is intentional -->

<!--

# PLEASE FOLLOW THE GUIDE BELOW

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes `[ ]` relevant to your *pull request* (like [x])
- Use *Preview* tab to see how your *pull request* will actually look like

-->

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

teika-kazura commented:
Your release_year is not always correct. For example [this episode](https://tver.jp/episodes/epbwpfftjg) was broadcast on "06-27", meaning June 27 in the last year.

I'm not sure of tver's date format. Perhaps if %m-%d looks of future, it's the last year? 

(This does not directly affect your code, but IIRC some epidsodes are labeled as "in year 2024", but were broadcast in late 2024.)

BTW I've just written a code to record both the broadcast date and the streaming closure date. [Here's the commit](https://github.com/teika-kazura/yt-dlp/commit/1642f15e3e70cef9809c75dae1d13a26fabbdf2e) If you want to use it, feel free to modify.

Thanks fireattack and bashonly. Best regards.

teika-kazura commented:
Furthermore, I think they should be called ``broadcast_date`` and _year - the names should describe what they are exacty. It seems that in yt_dlp ``release_date`` is important for many extractors, but these two are indeed different, as you correctly noticed. Put differnetly tver is such an extractor that the broadcast date matters more (for most watchers) than release_date, which is the date when that episode became available at tver.jp.

fireattack commented:
Well I think technically there are 3 date, if we use your definition:

1. broadcast date, there the original program was broadcasted on TV, probably the most useful one
2. 'release' date, the date where it was made available on TVer
3. the 'upload' date, which is the date where the video is uploaded to the brightcove which TVer uses. This date is often earlier than the actual broadcast/release date (sometimes weeks) because most of programs are pre-recorded/produced. This is an interesting info to know but ultimately not too useful for the typical watchers

Currently, TVer extractor only provide 3, `upload_date`, my PR is to add a `release_date`, (really is `broadcast_date`).

The reason I don't distinguish 1 & 2 is that 

1) I don't think you can get 2?
2) Most of programs are immediately available after aired on TV. There are exceptions, but in general true. So these two dates are mostly identical.

teika-kazura commented:
I found that the timestamp of the beginning of tver streaming is available as video_info['viewStatus']['startAt']. I'm sure of the meaning of this field, e.g. カンブリア宮殿 https://tver.jp/series/sre2549ef6 becomes available at tver one week later from the broadcast date, and 日経モープラFT https://tver.jp/series/sr6mw87agi, 8 hours later, which is not neglible as a news program.

So it is  actually meaningful to distinguish the broadcast date and the tver release date, because they can indeed differ,  and it is most natural to call the latter as 'release_date'. And all three of the broadcast date, the streaming beginning and end dattime are equally important. (For example you may want to sort by using one of them.)

Perhaps it might be good to add a comment  in the source explainig these dates; probably almost no one is aware of their exact definitions. (In particular that of the least interesting upload date, which surprised me.)

BTW it might be helpful to set isNHK = episode_content.get('isNHKContent'). (NHK *does* provide some programs at tver, e.g. https://tver.jp/series/sr5abiu37l ) I also considered if an episode is not broadcast on tv but available only in TVer, but such information seems to be lacking.

Thanks for your explanation. Now I think the dust has settled.

fireattack commented:
I've addressed these issues; however, as far as I know, we don't actually have a reserved `broadcast_date` field, so I'm not sure how to handle it. You do can use a custom field name in `-o`, at least:

```
>python -m yt_dlp "https://tver.jp/episodes/epbwpfftjg" --ignore-config -o "b_%(broadcast_date)s,r_%(release_date)s,u_%(upload_date)s.%(ext)s" --print filename
b_20240627,r_20240729,u_20240725.mp4 
```

Maybe a maintainer can chime in.

bashonly commented:
You could use `release_date` and `release_year`

fireattack commented:
@bashonly as discussed above we want to use `release_date` for the reliable metadata retrieved from API (the date a program is released on TVer), but still providing a more useful date (original TV air date) whenever it's possible. And I was wondering if it's ok to use a custom field for it. 

But if this deemed too complicated feel free to let me know.
